pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -592,9 +602,15 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 	// Is this a simple kubelet restart (scenario 2)? To distinguish, we use the information we got for runtime. If we are asked to allocate devices for containers reported
 	// running, then it can only be a kubelet restart. On node reboot the runtime and the containers were also shut down. Then, if the container was running, it can only be
 	// because it already has access to all the required devices, so we got nothing to do and we can bail out.
-	if !m.sourcesReady.AllReady() && m.isContainerAlreadyRunning(podUID, contName) {
-		klog.V(3).InfoS("container detected running, nothing to do", "deviceNumber", needed, "resourceName", resource, "podUID", podUID, "containerName", contName)
-		return nil, nil
+	if !m.sourcesReady.AllReady() {
+		if m.isContainerAlreadyRunning(podUID, contName) {
+			klog.V(3).InfoS("container detected running, nothing to do", "deviceNumber", needed, "resourceName", resource, "podUID", string(podUID), "containerName", contName)
+			return nil, nil
+		}
+		if isInitContainerCompleted {
```

Comment:
e.g. `if m.isInitContainerSucceeded(podUID, contname)` ...
This would look more clear than passing a boolean down in my opinion.